crawl-ref/source/dat/des/variable/mini_monsters.des

```diff
@@ -1229,6 +1229,76 @@ x<<xxxxxxxxxxxxxxxxxxcx
 x&&xxxxxxxxxxxxxxxxxxxx
 ENDMAP
 
+# I'd apologize for this pun, but I'm not actually sorry.
+NAME:    nicolae_half_adder
+TAGS:    transparent patrolling
+DEPTH:   D:2-6
+KMONS:   1 = adder
+KFEAT:   !O = c
+SHUFFLE: pqrs/O!!O/!O!O/!!O!
+SUBST:   pqrs = O
+: nsubst( "1 = " .. you.depth() .. "=1 / 1=1. / ." )
+TILE:    ! = DNGN_TRANSPARENT_STONE_WHITE
+TILE:    O = DNGN_TRANSPARENT_STONE_DARKGRAY
+COLOUR:  ! = white
+COLOUR:  O = magenta
+MAP
+.....................
+..p.........c.c....1.
+.pppppppppppcpcc..r..
+..p.p.1..1..1.%ccrrr.
+.1...p....qqcqcc..r..
+....1.p..q.1c.c..1...
+.......pq............
+.......qp............
+....1.q..p.1cccc.1...
+.1...q....ppccccc.s..
+..q.q.1..1..1.%ccsss.
+.qqqqqqqqqqqccccc.s..
+..q.........cccc...1.
+.....................
+ENDMAP
+
+# A similar idea didn't cut it as a Swamp rune vault, so let's resurrect it as
+# a lower-Dungeon vault. The gimmick, of course, is that each room has
+# 27 hydra heads, just arranged on a different number of hydra bodies.
+# Only places if this game doesn't have Swamp. This way, there's no incentive
+# to dive down through D in the hopes of finding this vault before the Swamp
+# end is generated to keep the Lernaean Hydra from being generated in the
+# rune vault.
+NAME:   nicolae_hydra_factorization
```

Comment:
This is a neat vault concept that seems like it probably does work. One suggestion I'd make is to rethink the way loot is placed. The contents of the first section are revealed, but the player has no idea what they're risking their lives for. The simplest solution here is to make the loot chamber have an `n` transparent wall on the far east that reveals the pile of loot available. This would also reveal that they'd eventually have to fight Learny, which seems fine and also funny for players who encounter the vault. They'd look at that loot pile safely from the outside but then they'd also look at a 27-headed boss hydra and think "...nope" or maybe make a bad decision. You could replace the final runed door with an ordinary door if you didn't want to reveal Lerny, since Lerny can't use doors. Personally it feels better to reveal the presence of Lerny ahead of time, but that's up to you.

Another loot suggestion that's less important than the one above, but something to consider. You could put smaller loot piles in every chamber in a way so that the pile was visible from the closest rune door. The first room might have only three items of lowest quality, perhaps simply gold, the next chamber might have 3 normal items, the next 3 star items? The Lerny chamber of course already has its own loot chamber with the main spoils and would need no adjustment. 

Doing this gives players an incentive to try the stages of the vault they think they can handle for a smaller but still notable reward. Otherwise they're really only going to attempt the entire vault in one go once they're strong enough to take on Lerny. At that point, the earlier chambers are just less interesting. Additionally you'd to have some consumable loot through use of `%` and `*` instead of all equipment and gold.

Reply:
Added a window on the far end so you can peep the loot, also place some $, %, and * in the previous three rooms so players can get some milder loot before the big guy.